azure_jumpstart_ag/artifacts/PowerShell/AgLogonScript.ps1

```diff
@@ -1834,6 +1817,53 @@ function Deploy-AIO {
     }
 }
 
+function Deploy-ESA {
+    # increase the maximum number of files
+    Invoke-AksEdgeNodeCommand -NodeType "Linux" -Command 'echo 'fs.inotify.max_user_instances = 1024' | sudo tee -a /etc/sysctl.conf && sudo sysctl -p'
+    
+    # Enable Open Service Mesh extension on the Arc-enabled cluster
+    Write-Host "[$(Get-Date -Format t)] INFO: Enabling Open Service Mesh on the Arc-enabled cluster" -ForegroundColor DarkGray
+    az k8s-extension create --resource-group $resourceGroup --cluster-name $arcClusterName --cluster-type connectedClusters --extension-type Microsoft.openservicemesh --scope cluster --name osm
+    
+    # Get the storage Account secret
+    $esaSecret = az storage account keys list --resource-group $resourceGroup -n $aioStorageAccountName --query "[0].value" -o tsv
+    
+    # Define names for ESA Yamls
+    $esaPVName = "esapv"
+    $esaPVCName = "esapvc"
+    $esaAppName = "esaapp"
+
+    # Inject params into the yaml file for PV
+    (Get-Content $esapvYaml ) -replace 'esaPVName', $esaPVName | Set-Content $esapvYaml
+    (Get-Content $esapvYaml ) -replace 'namespace', $aioNamespace | Set-Content $esapvYaml
+    (Get-Content $esapvYaml ) -replace 'esaContainerName', $stcontainerName | Set-Content $esapvYaml
+    (Get-Content $esapvYaml ) -replace 'esaSecretName', $ESAsecret | Set-Content $esapvYaml
+    
+    # Inject params into the yaml file for PVC
+    (Get-Content $esapvcYaml ) -replace 'esaPVCName', $esaPVCName | Set-Content $esapvcYaml
+    (Get-Content $esapvcYaml ) -replace 'namespace', $aioNamespace | Set-Content $esapvcYaml
+    (Get-Content $esapvcYaml ) -replace 'esaPVName', $esaPVName | Set-Content $esapvcYaml
+
+    # Inject params into the yaml file for ESA App
+    (Get-Content $esaappYaml ) -replace 'appname', $esaAppName | Set-Content $esaappYaml
+    (Get-Content $esaappYaml ) -replace 'namespace', $aioNamespace | Set-Content $esaappYaml
+    (Get-Content $esaappYaml ) -replace 'esaPVCName', $esaPVCName | Set-Content $esaappYaml
+
+    # Enable ESA extension on the Arc-enabled cluster
+    Write-Host "[$(Get-Date -Format t)] INFO: Enabling ESA on the Arc-enabled cluster" -ForegroundColor DarkGray
+    az k8s-extension create --resource-group $resourceGroup --cluster-name $arcClusterName --cluster-type connectedClusters --name hydraext --extension-type microsoft.edgestorageaccelerator --config-file $esapvJson
+    kubectl create secret generic -n $aioNamespace "$arcClusterName"-secret --from-literal=azurestorageaccountkey=$esaSecret --from-literal=azurestorageaccountname=$aioStorageAccountName
+    
+    Write-Host "[$(Get-Date -Format t)] INFO: Deploying PV on the Arc-enabled cluster" -ForegroundColor DarkGray
+    kubectl apply -f $esapvYaml
```

Comment:
Probably we need to check that the required storage for that PV is available in the node/cluster

Reply:
Agree, shall I suggest getting this automation running OK end to end and add those checks later?